initrd/bin/kexec-save-default

```diff
@@ -33,106 +33,213 @@ TMP_MENU_FILE="/tmp/kexec/kexec_menu.txt"
-			-o "$change_key_confirm" = "Y" ]; then
-			old_lvm_volume_group=""
-			if [ -r "$KEY_LVM" ]; then
-				old_lvm_volume_group=`cat $KEY_LVM` || true
-				old_key_devices=`cat $KEY_DEVICES \
-				| cut -d\  -f1 \
-				| grep -v "$old_lvm_volume_group" \
-				| xargs` || true
-			else
-				old_key_devices=`cat $KEY_DEVICES \
-				| cut -d\  -f1 | xargs` || true
-			fi
-
-			lvm_suggest="was '$old_lvm_volume_group'"
-			devices_suggest="was '$old_key_devices'"
-			save_key="y"
-		fi
-	fi
-
-	if [ "$save_key" = "y" ]; then
-		echo "+++ LVM volume groups (lvm vgscan): "
-		lvm vgscan || true
-
-		read \
-			-p "Encrypted LVM group? ($lvm_suggest): " \
-			lvm_volume_group
-
-		echo "+++ Block devices (blkid): "
-		blkid || true
-
-		read \
-			-p "Encrypted devices? ($devices_suggest): " \
-			key_devices
-
-		save_key_params="-s -p $paramsdev"
-		if [ -n "$lvm_volume_group" ]; then
-			save_key_params="$save_key_params -l $lvm_volume_group $key_devices"
-		else
-			save_key_params="$save_key_params $key_devices"
-		fi
-		echo "Running kexec-save-key with params: $save_key_params"
-		kexec-save-key $save_key_params \
-			|| die "Failed to save the disk key"
-	fi
+  if [ ! -s "$KEY_DEVICES" ]; then
+    read \
+      -n 1 \
+      -p "Do you wish to add a Disk Unlock Key to the TPM [Y/n]: " \
+      add_key_confirm
+    echo
+
+    if [ "$add_key_confirm" = "y" ] || [ "$add_key_confirm" = "Y" ] || [ -z "$add_key_confirm" ]; then
+      if [ "$num_lvm" -gt 1 ]; then
+        //untested
+        selected_lvmdev_not_existing=1
+        while [ $selected_lvmdev_not_existing -ne 0 ];do
+        {
+            read \
+              -p "Encrypted LVM group? choose between: $lvm_suggest: " \
+              lvm_volume_group
+
+            result=$(echo "$lvm_suggest" | grep -q "$lvm_volume_group") || selected_lvmdev_not_existing=1
+            if [ $? == 0 ]; then
+              selected_lvmdev_not_existing=0
+            fi
+        };done  
+      elif [ "$num_lvm" -eq 1 ]; then
+        echo "Single Encrypted LVM group found at $lvm_suggest. Using it."
+        lvm_volume_group=$lvm_suggest
+      else
+        echo "No encrypted LVM Group found."
+      fi
+
+      if [ "$num_devices" -gt 1 ]; then
+        selected_luksdev_not_existing=1
+        while [ $selected_luksdev_not_existing -ne 0 ];do
+        {
+          read \
+          -p "Encrypted devices? (choose between: $devices_suggest): " \
+          key_devices
+
+          result=$(echo "$devices_suggest" | grep -q "$key_devices") || selected_luksdev_not_existing=1
+          if [ $? == 0 ]; then
+            selected_luksdev_not_existing=0
+          fi
+        };done
+      elif [ "$num_devices" -eq 1 ]; then
+        echo "Single Encrypted Disk found at $devices_suggest. Using it."
+        key_devices=$devices_suggest
+      else
+        echo "No encrypted devices found."
+      fi
+
+    save_key="y"
+    else
+      die "User chose to not add Disk Unlock Key to the TPM"
```

Comment:
@kylerankin : so the culprit is here. Changing it to warning, while moving [the whole LVM and encrypted disk logic on top](https://github.com/osresearch/heads/pull/551/files#diff-08c6ecea92d88d78364ed40007bddc2aR165) reduce duplication of code and permits to select a default without forcing it to be saved into TPM. Testing

Reply:
@kylerankin : [while moving this part down](https://github.com/osresearch/heads/pull/551/files#diff-08c6ecea92d88d78364ed40007bddc2aR209), outside of the if then else part. Testing

Reply:
@kylerankin : changed to logic to address and try to conceal to your [concerns here](https://github.com/osresearch/heads/pull/551#discussion_r277890574)